tools/system_libs.py

```diff
@@ -1376,6 +1376,10 @@ class libcxxabi(NoExceptLibrary, MTLibrary, DebugLibrary):
     ]
   includes = ['system/lib/libcxx/src']
 
+  def __init__(self, **kwargs):
+    super().__init__(**kwargs)
+    self.is_debug |= settings.EXCEPTION_STACK_TRACES
```

Comment:
Using the debug version will affect `NDEBUG`. Maybe for libcxxabi that's not significant, though? But I think it's worth measuring code size just to make sure.

(If that ends up an issue, we could maybe have a separate library for the function that calls out to JS, and link it first so it overrides part of libcxxabi, but hopefully that's not necessary.)

Reply:
We already use the debug version of libc++abi when `ASSERTIONS` is set. This was enabled in #17979. Having a debug version of libc++abi also has other advantages, such as, it shows abort error messages, which the release version doesn't. I don't think people would mind much about the small code increase when they set `EXCEPTION_STACK_TRACES` to true..? We also discussed whether enabling the debug version of libc++abi was worth it last year in our team chat and decided it was.

If we want to avoid the code size increase for libc++abi when `ASSERTIONS` is not set but `EXCEPTION_STACK_TRACES` is set, I think we need to come up with a separate preprocessor keyword for the code snippet included in https://github.com/emscripten-core/emscripten/pull/18642/files#r1093681153.

FWIW, the code sizes for `wasm32-emscripten` in my emscripten cache are:
```console
aheejin@aheejin:~/emscripten/cache/sysroot/lib/wasm32-emscripten$ ls libc++abi* -al
-rw-r----- 1 aheejin primarygroup  935626 Feb  1 13:33 libc++abi.a
-rw-r----- 1 aheejin primarygroup 1295312 Feb  1 13:33 libc++abi-debug.a
-rw-r----- 1 aheejin primarygroup 1376762 Feb  1 13:34 libc++abi-debug-except.a
-rw-r----- 1 aheejin primarygroup 1310656 Feb  1 13:33 libc++abi-debug-mt.a
-rw-r----- 1 aheejin primarygroup 1396908 Feb  1 13:34 libc++abi-debug-mt-except.a
-rw-r----- 1 aheejin primarygroup 1304380 Feb  1 13:33 libc++abi-debug-mt-noexcept.a
-rw-r----- 1 aheejin primarygroup 1289086 Feb  1 13:33 libc++abi-debug-noexcept.a
-rw-r----- 1 aheejin primarygroup 1310656 Feb  1 13:33 libc++abi-debug-ww.a
-rw-r----- 1 aheejin primarygroup 1396908 Feb  1 13:34 libc++abi-debug-ww-except.a
-rw-r----- 1 aheejin primarygroup 1304380 Feb  1 13:33 libc++abi-debug-ww-noexcept.a
-rw-r----- 1 aheejin primarygroup 1016138 Feb  1 13:33 libc++abi-except.a
-rw-r----- 1 aheejin primarygroup  951196 Feb  1 13:33 libc++abi-mt.a
-rw-r----- 1 aheejin primarygroup 1036278 Feb  1 13:34 libc++abi-mt-except.a
-rw-r----- 1 aheejin primarygroup  946236 Feb  1 13:33 libc++abi-mt-noexcept.a
-rw-r----- 1 aheejin primarygroup  930950 Feb  1 13:33 libc++abi-noexcept.a
-rw-r----- 1 aheejin primarygroup  951196 Feb  1 13:33 libc++abi-ww.a
-rw-r----- 1 aheejin primarygroup 1036278 Feb  1 13:34 libc++abi-ww-except.a
-rw-r----- 1 aheejin primarygroup  946236 Feb  1 13:33 libc++abi-ww-noexcept.a
```

Reply:
It might be fine to increase the code size here. But some of the motivation for this new option was to get stack traces without the extra overhead and size of all of `ASSERTIONS`. Anyhow, this is still a big improvement over `ASSERTIONS` so I'd be ok with it with a TODO to investigate further optimizations in `system_libs.py`, if we find the need for that later. (From the `.a` files the debug version does look larger, but that might not matter after linking or in comparison to other libraries I guess.)

Reply:
Added the TODO.